Issue #15713: updated JavadocParagraph's input files based on issue description

Zopsss opened the pull request:
fixes #15713

Zopsss commented on config/checkstyle-resources-suppressions.xml:
```diff
@@ -256,10 +256,6 @@
     files="[\\/]test[\\/]resources[\\/]com[\\/]puppycrawl[\\/]tools[\\/]checkstyle[\\/]checks[\\/]javadoc[\\/]javadocmethod[\\/]InputJavadocMethodPublicOnly1\.java"/>
   <suppress checks="FileLength"
     files="[\\/]test[\\/]resources[\\/]com[\\/]puppycrawl[\\/]tools[\\/]checkstyle[\\/]checks[\\/]javadoc[\\/]javadocmethod[\\/]InputJavadocMethodProtectedScopeJavadoc\.java"/>
-  <suppress checks="FileLength"
-    files="[\\/]test[\\/]resources[\\/]com[\\/]puppycrawl[\\/]tools[\\/]checkstyle[\\/]checks[\\/]javadoc[\\/]javadocparagraph[\\/]InputJavadocParagraphIncorrect\.java"/>
-  <suppress checks="FileLength"
-    files="[\\/]test[\\/]resources[\\/]com[\\/]puppycrawl[\\/]tools[\\/]checkstyle[\\/]checks[\\/]javadoc[\\/]javadocparagraph[\\/]InputJavadocParagraphIncorrect2\.java"/>
```
separated InputJavadocParagraphIncorrect2 in 2 files a( InputJavadocParagraphIncorrect2 & InputJavadocParagraphIncorrect3 )

InputJavadocParagraphIncorrect became small so didn't needed to split it

Zopsss commented:
https://github.com/checkstyle/checkstyle/actions/runs/11085472433/job/30801929655?pr=15717#step:7:23

pitest is indicating JavadocTokenTypes.EOF is safe to delete. @romani

romani commented on src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphIncorrect3.java:
```diff
@@ -0,0 +1,52 @@
+         * Double newline.
+         *
+         *
+         * Some Javadoc. //DOUBLE WARN AT TWO PREVIOUS LINES
```
Please remove trailing comment from javadoc ,

Zopsss replied:
done

romani reviewed (changes requested):
We need green ci

romani commented on src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule712paragraphs/InputFormattedIncorrectJavadocParagraph.java:
```diff
@@ -3,7 +3,7 @@
  * Some Javadoc.
  *
- * <p>/^ WARN/ Some Javadoc.
+ * <p>Some Javadoc.
```
Please change `Some Javadoc.`  to ` some paragraph`.
And summary of javadoc to "Some summary.`.
To make it clear what is what in javadoc content.

To hint user that it is paragraph text.
Please apply to all.

Zopsss replied:
done

romani reviewed (changes requested):
Items

romani commented:
```
+  <mutation unstable="false">
+    <sourceFile>JavadocParagraphCheck.java</sourceFile>
+    <mutatedClass>com.puppycrawl.tools.checkstyle.checks.javadoc.JavadocParagraphCheck</mutatedClass>
+    <mutatedMethod>isImmediatelyFollowedByText</mutatedMethod>
+    <mutator>org.pitest.mutationtest.engine.gregor.mutators.NonVoidMethodCallMutator</mutator>
+    <description>removed call to com/puppycrawl/tools/checkstyle/api/DetailNode::getType</description>
+    <lineContent>return nextSibling.getType() == JavadocTokenTypes.EOF</lineContent>
+  </mutation>
+
+  <mutation unstable="false">
+    <sourceFile>JavadocParagraphCheck.java</sourceFile>
+    <mutatedClass>com.puppycrawl.tools.checkstyle.checks.javadoc.JavadocParagraphCheck</mutatedClass>
+    <mutatedMethod>isImmediatelyFollowedByText</mutatedMethod>
+    <mutator>org.pitest.mutationtest.engine.gregor.mutators.RemoveConditionalMutator_EQUAL_IF</mutator>
+    <description>removed conditional - replaced equality check with true</description>
+    <lineContent>return nextSibling.getType() == JavadocTokenTypes.EOF</lineContent>
+  </mutation>
```

this means we lost test case (some code in Input) that kept EOF meanigful.

Zopsss commented:
> please recheck that we still have unrealistic javadoc `...Javadoc.<p>*/`

we have this suppression:

https://github.com/checkstyle/checkstyle/blob/a8db72d7330b43e8a9ad08acdcd8f71fd1fe8e17/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphIncorrect.java#L47-L54

romani commented:
I am confused, we just changed input files,   and it means we just lost test case that covered EOF.
Please revert charges on your local by files to catch what file covered EOF. And later on revert charges in file to find exact javadoc that use to cover EOF.

Zopsss commented on src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphIncorrect3.java:
```diff
@@ -0,0 +1,52 @@
+         *  <p>  Some Javadoc.
+         *
+         * @see <a href="http://www.gwtproject.org/doc/latest/DevGuideOrganizingProjects.html#DevGuideModules">
+         *     Documentation about <p> GWT emulated source</a>
```
done

Zopsss commented:
> I am confused, we just changed input files, and it means we just lost test case that covered EOF. Please revert charges on your local by files to catch what file covered EOF. And later on revert charges in file to find exact javadoc that use to cover EOF.

I added this test case again:

https://github.com/checkstyle/checkstyle/blob/5b9c6b76736d9b78bae2735cc4846b00be414f38/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphIncorrect2.java#L99-L104

without `EOF` test started failing, I guess we will have green ci this time.

romani reviewed (approved):
Awesome step forward.

@rdiachenko , I am merging this by single review, it is just inputs clean up
